test/test_model.py

```diff
@@ -167,3 +180,82 @@ def test_config(self):
         with self.assertRaises(TypeError):
             # Confirm that we cannot modify config values.
             self.model.config['foo'] = 'bar'
+
+    def test_is_leader(self):
+        self.backend = op.model.ModelBackend()
+        self.model = op.model.Model('myapp/0', ['db0', 'db1', 'db2'], self.backend)
+
+        def check_remote_units():
+            fake_script(self, 'relation-ids',
+                        '''
+                        case $1 in
+                            db1) echo '["db1:4"]' ;;
+                            *) echo '[]'
+                        esac
+                        ''')
+
+            fake_script(self, 'relation-list',
+
+                        '''
+                        fail_not_found() {
+                            1>&2 echo "ERROR invalid value \"$1\" for option -r: relation not found"
+                            exit 2
+                        }
+
+                        case $2 in
+                        4) echo '["remoteapp1/0", "remoteapp1/1"]' ;;
+                        *) fail_not_found $2 ;;
+                        esac
+                        ''')
+
+            # Cannot determine leadership for remote units.
+            for u in self.model.get_relation('db1').units:
+                with self.assertRaises(RuntimeError):
+                    u.is_leader
+
+        fake_script(self, 'is-leader', 'echo -n true')
+        self.assertTrue(self.model.unit.is_leader)
+
+        check_remote_units()
+
+        self.backend = op.model.ModelBackend()
+        self.model = op.model.Model('myapp/0', ['db0', 'db1', 'db2'], self.backend)
+
+        fake_script(self, 'is-leader', 'echo -n false')
+        self.assertFalse(self.model.unit.is_leader)
+
+        check_remote_units()
```

Comment:
One more detail: we can use fake_script_calls here to confirm the parameters used.